modular_zubbers/code/modules/antagonists/traitor/uplink/uplink_items/syndimaid/clothes.dm

```diff
@@ -16,44 +16,34 @@
 	armor_type = /datum/armor/clothing_under/syndimaid
 	body_parts_covered = CHEST|GROIN|LEGS
 	resistance_flags = parent_type::resistance_flags | FIRE_PROOF
+	/// who we are currently locking from wearing other outer clothing
+	var/datum/weakref/wear_locked
 
-	var/is_worn = FALSE // This is to avoid any jank with the equip chain that I know exists
-
-/obj/item/clothing/under/syndicate/skyrat/maid/armored/Initialize(mapload)
-	. = ..()
-	RegisterSignal(src, COMSIG_ITEM_POST_EQUIPPED, PROC_REF(check_equip))
-
-/obj/item/clothing/under/syndicate/skyrat/maid/armored/Destroy()
+/obj/item/clothing/under/syndicate/skyrat/maid/armored/equipped(mob/living/user, slot)
 	. = ..()
-	UnregisterSignal(src, COMSIG_ITEM_POST_EQUIPPED)
-
-/obj/item/clothing/under/syndicate/skyrat/maid/armored/mob_can_equip(mob/living/user, slot, disable_warning, bypass_equip_delay_self, ignore_equipped, indirect_action)
 	if(slot == ITEM_SLOT_ICLOTHING)
-		var/mob/living/carbon/human/human_wearer = user
-		if(human_wearer.wear_suit)
-			return FALSE
-	. = ..()
+		toggle_equip_lock(TRUE, user)
 
-/obj/item/clothing/under/syndicate/skyrat/maid/armored/proc/check_equip(datum/source, mob/user, slot, initial)
-	SIGNAL_HANDLER
-	if(slot == ITEM_SLOT_ICLOTHING)
-		RegisterSignal(user, COMSIG_HUMAN_EQUIPPING_ITEM, PROC_REF(block_equips))
-		RegisterSignal(user, COMSIG_MOB_UNEQUIPPED_ITEM, PROC_REF(remove_block))
-		is_worn = TRUE
-	else if(is_worn)
-		UnregisterSignal(user, list(COMSIG_HUMAN_EQUIPPING_ITEM, COMSIG_MOB_UNEQUIPPED_ITEM))
-		is_worn = FALSE
-
-/obj/item/clothing/under/syndicate/skyrat/maid/armored/proc/remove_block(datum/source, obj/item, force, newloc, no_move, invdrop, silent)
-	SIGNAL_HANDLER
-	if(is_worn)
-		UnregisterSignal(source, list(COMSIG_HUMAN_EQUIPPING_ITEM, COMSIG_MOB_UNEQUIPPED_ITEM))
-		is_worn = FALSE
-
-/obj/item/clothing/under/syndicate/skyrat/maid/armored/proc/block_equips(datum/source, mob/target, slot)
-	SIGNAL_HANDLER
-	if(slot == ITEM_SLOT_OCLOTHING)
-		return COMPONENT_BLOCK_EQUIP
+/obj/item/clothing/under/syndicate/skyrat/maid/armored/dropped(mob/living/user)
+	. = ..()
+	var/old_wear_locked = wear_locked?.resolve()
+	if(!old_wear_locked && old_wear_locked != user)
+		return
+	toggle_equip_lock(FALSE, user)
+
+/obj/item/clothing/under/syndicate/skyrat/maid/armored/proc/toggle_equip_lock(lock = TRUE, mob/living/user)
+	var/mob/living/carbon/human/human_wearer = user
+	if(!istype(human_wearer) && !human_wearer?.dna?.species)
```

Comment:
same as above, if human_wearer doesn't exist then surely dna and species wont

Reply:
excess paranoia of me trying to cover every base